.github/workflows/exctarct-en-lang.yml

```diff
@@ -0,0 +1,69 @@
+name: Extract menu strings to en.yml
+
+on:
+  push:
+    branches: [ main ]
+    paths:
+      - 'mkdocs.yml'
+
+permissions:
+  contents: write
+  actions: write
+
+jobs:
+  update-en-yml:
+    name: Process changes in English menu items
+    runs-on: ubuntu-latest
+    outputs:
+      has_changes: ${{ steps.check_changes.outputs.has_changes }}
+    env:
+      MKDOCS_FILE: "mkdocs.yml"
+      TARGET_FILE: "./docs/en/en.yml"
+
+    steps:
+      - name: ⬇️ Checkout repository
+        uses: actions/checkout@v4
+
+      - name: 🧰 Set up yq
+        uses: mikefarah/yq@v4
+
+      - name: 📝 Update en.yml
+        run: |
+          echo "Extracting English menu items from ${MKDOCS_FILE} to ${TARGET_FILE}"
+
+          TMP_FILE=$(mktemp)
+
+          yq eval '(.plugins[].i18n.languages[] | select(.locale == "en")) | del(.default)' "$MKDOCS_FILE" > "$TMP_FILE"
+
+          yq eval --inplace "
+            (.en.plugins[].i18n.languages[] | select(.locale == \"en\")) = load (\"$TMP_FILE\")
+          " "$TARGET_FILE"
+
+          rm "$TMP_FILE"
+
+      - name: Check if something has been changed in en.yaml
+        id: check_changes
+        run: |
+          if git diff-index --quiet HEAD -- "${TARGET_FILE}"; then
+            echo "No changes detected in ${TARGET_FILE}"
+            echo "has_changes=false" >> $GITHUB_OUTPUT
+          else
+            echo "Changes detected in ${TARGET_FILE}"
+            echo "Git diff output:"
+            git diff HEAD -- "${TARGET_FILE}" || true
+            echo "has_changes=true" >> $GITHUB_OUTPUT
+          fi
+
+      - name: Commit changes
+        if: steps.check_changes.outputs.has_changes == 'true'
+        run: |
+          git config user.name "github-actions[bot]"
+          git config user.email "github-actions[bot]@users.noreply.github.com"
+          git add "${TARGET_FILE}"
+          git commit -m "Changes in ${TARGET_FILE}"
```

Comment:
> The bot of the translation platform (like Transifex) monitors changes in the repo and synchronises them with the platform

Besides Transifex, Weblate and Crowdin also support this kind of workflow. I think the only one that requires a different workflow would be Translatewiki.net (which happens to be the TMS that openstreetmap-website uses). As long as we’re OK with ruling out Translatewiki.net in favor of a different hosted service, this is a pretty clean workflow for us to use for this relatively small site.

Reply:
Unfortunately, I don't have any experience of running a translation project on Translatewiki.net, but these Actions are not platform-specific. They are only intended to extract the original English source for further translation and then integrate the translations into a shared mkdocs.yml file.